src/miniscript/types/extra_props.rs

```diff
@@ -117,6 +117,15 @@ pub struct ExtData {
     pub stack_elem_count_sat: Option<usize>,
     /// The worst case number of stack elements for dissatisfying this Miniscript fragment.
     pub stack_elem_count_dissat: Option<usize>,
+    /// Maximum size, in bytes, of a satisfying witness. First elements is the cost for the
+    /// witness stack, the second one is the cost for scriptSig.
+    /// All signatures are assumed to be 73 bytes in size, including the
+    /// length prefix (segwit) or push opcode (pre-segwit) and sighash
+    /// postfix.
+    pub max_sat_size: Option<(usize, usize)>,
```

Comment:
Putting out some random thought here. I have not completely evaluated the solution. Since we force the user to choose Miniscript Context (via descriptor) or by explicitly specifying it, we have the information whether the Miniscrpit is Segwit or not. Would it be possible to have Ctx information in Property and a function one_cost() in ScriptContext trait? That should simplify this code.

Reply:
Yeah, it would greatly simplify this commit full of nested `map()` and `and_then()`.

Familiarizing with the internals i thought that we could start stuffing quite a lot of things in the Ctx :sweat_smile:

Reply:
I dropped a fixme for this :/